SCClassLibrary/Common/UnitTesting/UnitTest.sc

```diff
@@ -189,6 +189,23 @@ UnitTest {
 		}
 	}
 
+	assertException { | func, errorClass, message, report = true, onFailure, details |
+		var resultingError;
+		errorClass = errorClass.asClass;
+		func.try { |error|
+			if(error.isKindOf(errorClass)) { resultingError = error };
+		};
+		this.assertEquals(resultingError.class, errorClass, "resultingError should match the given errorClass");
+		this.assert(resultingError.notNil, message, report, onFailure, details)
```

Comment:
this breaks the abstraction - basically, users should expect that a call to an assert-type method should result in exactly one pass/fail message. We already have a pattern established by `assertEquals` of adding additional information about failures to the `details` argument.

Reply:
"this breaks the abstraction" is not really true. I know you'd like to establish the convention that there should be only one assert per method.

I have other things to do now, I'll see when I get round to it.

Reply:
This isn't really about one assert per method, it's about one assert per "assert."

Reply:
Do you mind if I push to this? There are now at least 2 open PRs that would benefit from this.